utils/src/main/scala/za/co/absa/enceladus/utils/validation/field/ScalarFieldValidator.scala

```diff
@@ -24,7 +24,9 @@ import scala.util.Try
 /**
   * Scalar types schema validation against default value
   */
-object ScalarFieldValidator extends FieldValidator {
+object ScalarFieldValidator extends ScalarFieldValidator
+
+class ScalarFieldValidator extends FieldValidator {
```

Comment:
I see that you turned this into a class so that it could be inherited by `FractionalFieldValidator`, shouldn't `ScalarFieldValidator` be declared `abstract` since you want the companion `object` to be the only instance?

Reply:
Class can be freely instantiated, not that there is a much point of it, the companion object implementing the class doess all the class offers.
But the restriction via `abstract` declaration seems unnecessarily restrictive.